artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/broker/AMQPLargeMessage.java

```diff
@@ -146,16 +147,6 @@ private AMQPLargeMessage(final AMQPLargeMessage copy,
       setMessageID(newID);
    }
 
-   public void openLargeMessage() throws Exception {
```

Comment:
The 'getData()' method appears to depend on this method having created "parsingData". Though I admit this method does seem ununused, getData() seems to be used a lot. I thus wonder why this seems unused

Reply:
@clebertsuconic any thoughts here?

Reply:
@gemmellr / @franz1981 I remember using this at some point.. I don't know where it stopped using it. 

I didn't find any usage of it for sure.

Reply:
at some point I started using getWholeBodyReader on Core to AMQP conversion (from LargeBody), and BodyReader on the AMQP Message. 

If the test suite is passing, this is fine as the test suite will cover conversion of large messages between core and AMQP.

Reply:
There is a buffer that this opens. That buffer is still being referenced in getData(), and throws but only if it isnt set, which it never will be if this isnt used. One of these situations seems like it has to be wrong.

Reply:
Waiting @clebertsuconic answer to know what to remove here

Reply:
getData is used on conversion.. if you send an AMQP Large Message and receive it on core.. getData will be used to read the whole thing into memory.

Reply:
but getData() is not opening the largeMessage as far as I know... it's just using the parsing data... let me take a look

Reply:
It isnt opening it, true. It relied on the the 'now unused' openLargeMessage() and closeLargeMessage() methods just removed by this PR to open and close the buffer it uses. As that no longer happens, nothing will, and getData() will throw if ever actually called.